src/containers/NonBitGoRecoveryCoin/SolanaForm.tsx

```diff
@@ -0,0 +1,191 @@
+import { Form, FormikHelpers, FormikProvider, useFormik } from 'formik';
+import { Link } from 'react-router-dom';
+import * as Yup from 'yup';
+import {
+  Button,
+  FormikSelectfield,
+  FormikTextarea,
+  FormikTextfield,
+  Icon,
+  Notice,
+} from '~/components';
+
+const validationSchema = Yup.object({
+  backupKey: Yup.string().required(),
+  bitgoKey: Yup.string().required(),
+  krsProvider: Yup.string()
+    .oneOf(['keyternal', 'bitgoKRSv2', 'dai'])
+    .label('Key Recovery Service'),
+  publicKey: Yup.string().required(),
+  recoveryDestination: Yup.string().required(),
+  scan: Yup.number().required(),
+  userKey: Yup.string().required(),
+  walletPassphrase: Yup.string().required(),
+})
+  .required()
+  .shape(
+    {
+      publicKey: Yup.string().when('secretKey', {
+        is: (val: string) => !!val,
+        then: Yup.string().required(
+          'A public key must be provided with a secret key'
+        ),
+      }),
+      secretKey: Yup.string().when('publicKey', {
+        is: (val: string) => !!val,
+        then: Yup.string().required(
+          'A secret key must be provided with a public key'
+        ),
+      }),
+    },
+    [['secretKey', 'publicKey']]
+  );
```

Comment:
does this make sense to you? does it do what you want? because I don't understand what the intent with this is.